packages/amplify-category-storage/src/provider-utils/awscloudformation/service-walkthroughs/s3-questions.ts

```diff
@@ -92,7 +92,7 @@ export async function askBucketNameQuestion(context: $TSContext, defaultValues:
   const message = 'Provide bucket name:';
   const onErrorMsg =
     'Bucket name can only use the following characters: a-z 0-9 - and should have minimum 3 character and max of 47 character';
-  const validator = and([alphanumeric(), minLength(3), maxLength(47)], onErrorMsg);
+  const validator = and([matchRegex(onErrorMsg, /^[a-z0-9-]+$/ ),minLength(3), maxLength(47)], onErrorMsg);
```

Comment:
nit: there seems to be an extra space and missing space here.
```suggestion
  const validator = and([matchRegex(onErrorMsg, /^[a-z0-9-]+$/), minLength(3), maxLength(47)], onErrorMsg);
```